Source/Classes/SLKTextViewController.m

```diff
@@ -322,15 +322,24 @@ - (SLKTextInputbar *)textInputbar
     return _textInputbar;
 }
 
-- (SLKTypingIndicatorView *)typingIndicatorView
+- (UIView <SLKTypingIndicatorProtocol> *)typingIndicatorProxyView
 {
-    if (!_typingIndicatorView)
+    if (!_typingIndicatorProxyView)
     {
-        _typingIndicatorView = [SLKTypingIndicatorView new];
-        _typingIndicatorView.translatesAutoresizingMaskIntoConstraints = NO;
-        _typingIndicatorView.canResignByTouch = NO;
+        Class class = self.typingIndicatorViewClass ? : [SLKTypingIndicatorView class];
+        
+        _typingIndicatorProxyView = [[class alloc] init];
+        _typingIndicatorProxyView.translatesAutoresizingMaskIntoConstraints = NO;
+        _typingIndicatorProxyView.hidden = YES;
+        
+        [_typingIndicatorProxyView addObserver:self forKeyPath:NSStringFromSelector(@selector(isVisible)) options:NSKeyValueObservingOptionNew context:nil];
     }
-    return _typingIndicatorView;
+    return _typingIndicatorProxyView;
+}
+
+- (SLKTypingIndicatorView *)typingIndicatorView
+{
+    return (SLKTypingIndicatorView *)self.typingIndicatorProxyView;
```

Comment:
This will probably crash if you try to do something with the returned value, and it isn't really a kind of `SLKTypingIndicatorView`. You may want to return `nil` in this case?